### PR TITLE
Performance improvements v1

### PR DESCRIPTION
**Pull request by tourdedave**
re: [Trello 403](https://trello.com/c/EhPoivbv)

When running TestCafe concurrently with ClientFunctions, there appears to be a common performance penalty (seemingly regardless of the page's payload size extracted with dom-snapshot). This PR is a means to try and pay down some of that tax.

The current approach taken is to pass `dontFetchResources` to dom-snapshot, and let the vg-client pick up the slack.

This isn't a home-run performance improvement, but based on anecdotal benchmarks, it shows the potential to improve overall concurrent test execution times by ~10%.

### CODE REVIEW

File: src/makeClientFunctionWrapper.js [hunk 1]
```diff
@@ -38,17 +47,21 @@ function makeClientFunctionWrapper({
     );
 
     return async t => {
+      const testName = t.testRun.test.name;
```
Comment (amitzur) [inline src/makeClientFunctionWrapper.js:50]: I just want to verify that this is supported in all cases and all the TestCafe versions that we support

File: src/makeClientFunctionWrapper.js [hunk 2]
```diff
@@ -18,16 +18,25 @@ function makeClientFunctionWrapper({
 }) {
   return async function(browserFunction, dependencies = {}) {
     const getResultSize = clientFunctionExecuter(
-      () =>
-        browserFunction().then((result = {}) => {
+      () => {
+        // eslint-disable-next-line no-undef
+        return browserFunction(functionArgs).then((result = {}) => {
```
Comment (amitzur) [inline src/makeClientFunctionWrapper.js:23]: why isn't this `dependencies.functionArgs`? How does this work if it's undefined?

File: src/eyes.js [hunk 3]
```diff
@@ -103,7 +106,15 @@ class Eyes {
 
   async _processPage(t) {
     if (!this._processPageClientFunction) {
-      this._processPageClientFunction = await this._clientFunctionWrapper(processPageAndSerialize);
+      this._processPageClientFunction = await this._clientFunctionWrapper(processPageAndSerialize, {
+        functionArgs: [
+          undefined, // will default processPage to document
+          {
+            dontFetchResources: !!process.env.APPLITOOLS_DISABLE_BROWSER_FETCHING,
```
Comment (amitzur) [inline src/eyes.js:113]: Ideally this should be passed by the user in `eyes.checkWindow` arguments. I will continue the discussion with product in Trello

File: src/makeClientFunctionWrapper.js (hunk 1 above)
Comment (tourdedave) [reply src/makeClientFunctionWrapper.js:50]: Good idea. In the interim, we could make this more defensive (e.g., `t.testRun && t.testRun.test && t.testRun.test.name ? t.testRun.test.name || ''`

File: src/makeClientFunctionWrapper.js (hunk 2 above)
Comment (tourdedave) [reply src/makeClientFunctionWrapper.js:23]: `dependencies` doesn't exist in this context, just its destructured contents.

File: src/eyes.js (hunk 3 above)
Comment (tourdedave) [reply src/eyes.js:113]: Seems like we'd want an either/or option (global or granular). Let me know where y'all land.

File: src/makeClientFunctionWrapper.js (hunk 2 above)
Comment (amitzur) [reply src/makeClientFunctionWrapper.js:23]: I must be missing something. `dependencies` is the second parameter to the wrapping function in line 19. Is there some magic going on related to TestCafe?

**Comment by tourdedave** [comment]
Yes.

**Pull request merged**